src/store/sagas/currentUserJoinRoom.ts

```diff
@@ -0,0 +1,19 @@
+import { PayloadAction } from '@reduxjs/toolkit';
+import { put, take } from 'redux-saga/effects';
+import { currentUserJoinRoom, ICurrentUserJoinRoomData, setDisplayName, setRoomCode } from '../main';
+
+export function* watchCurrentUserJoinRoom() {
+  while (true) {
+    const {
+      payload: {
+        user: {
+          name,
+          room
+        }
+      }
+    }: PayloadAction<ICurrentUserJoinRoomData> = yield take(currentUserJoinRoom.toString());
```

Comment:
Need the typing on the var result of `yield` as it is unable to infer the correct type.